main.py

```diff
@@ -128,6 +128,12 @@ def pyro_model(x, i, y):
 
 
 def localnews(INFERENCE):
+    device = torch.device('cpu')
+    torch.set_default_tensor_type(torch.DoubleTensor)
+    if torch.cuda.is_available():
+        device = torch.device('cuda')
+        torch.set_default_tensor_type(torch.cuda.DoubleTensor)
```

Comment:
This defines all tensors as CUDA based doubles. Tensors are all moved to CPU later before visualization.